extensions/jdbc/jdbc-postgresql/runtime/src/main/java/io/quarkus/jdbc/postgresql/runtime/PostgreSqlServiceBindingConverter.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.jdbc.postgresql.runtime;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBinding;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConfigSource;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter;
+
+public class PostgreSqlServiceBindingConverter implements ServiceBindingConverter {
+
+    private static final Logger log = Logger.getLogger(ServiceBinding.class);
+
+    @Override
+    public Optional<ServiceBindingConfigSource> convert(List<ServiceBinding> serviceBindings) {
+        Optional<ServiceBinding> matchingByType = ServiceBinding.singleMatchingByType("postgresql", serviceBindings);
+        if (!matchingByType.isPresent()) {
+            return Optional.empty();
+        }
+
+        Map<String, String> properties = new HashMap<>();
+        ServiceBinding binding = matchingByType.get();
+
+        String username = binding.getProperties().get("username");
+        if (username != null) {
+            properties.put("quarkus.datasource.username", username);
+        } else {
+            log.debug("Property 'username' was not found");
+        }
+        String password = binding.getProperties().get("password");
+        if (password != null) {
+            properties.put("quarkus.datasource.password", password);
+        } else {
+            log.debug("Property 'password' was not found");
+        }
+        String host = binding.getProperties().get("host");
+        String port = binding.getProperties().get("port");
+        String database = binding.getProperties().get("database");
+        if ((host != null) && (database != null)) {
+            String portPart = "";
+            if (port != null) {
+                portPart = ":" + port;
+            }
+            properties.put("quarkus.datasource.jdbc.url", String.format("jdbc:postgresql://%s%s/%s", host, portPart, database));
+        } else {
+            log.debug("One or more of 'host' or 'database' properties were not found");
+        }
+        return Optional.of(new ServiceBindingConfigSource("postgresql-k8s-service-binding-source", properties));
```

Comment:
I'm wondering how we should map other properties available in the ServiceBinding? For instance if I want to use the `url` directly, should that be in a file `jdbc.url` and we add the required prefixes?

Reply:
I am not sure that is even allowed by the Service Binding spec. I basically followed the Spring Cloud code here and did what they did (the author is also one of the lead authors of the spec - having also been a part of the previous Cloud Foundry specs, so I pretty much consider it as a sort of reference implementation :))

Reply:
What I understood from the spec is that there was some well known entries, but you can add any entry that you like. In that case, how do we access it?

Reply:
I think we need more experience with the actual usage of the service binding before we commit to something.

Reply:
there should be two kind of mappings of the service binding options imo:

1) just a raw mapping of the key/bindings into its own "property hiearchy" which can then be used by user to map whatever they want/need - we shouldn't alter the hierarchy here imo.

2) additional extra step of iterating of known entries (i.e. postgres, kafkaxyz, etc.) which would map those known bidnings into quarkus specific extensions. This can include custom mappings/manipulations.

Reply:
Yup, we have that now - see Roberto's other comment which I addressed.